src/librustc_errors/emitter.rs

```diff
@@ -703,6 +703,21 @@ impl EmitterWriter {
         }
     }
 
+    fn msg_with_padding(&self, msg: &str, padding: usize) -> String {
+        let padding = (0..padding)
+            .map(|_| " ")
+            .collect::<String>();
```

Comment:
If I get it right, it will allocate every time the method is called.
What do you think about adding a function
```rust
fn add_padding(msg: &mut String, padding: usize) {
    for _ in 0..padding {
        msg.push(' ');
    }
}
```
and using it instead of pushing padding.